ocaml/xapi/create_misc.ml

```diff
@@ -387,7 +387,30 @@ let make_packs_info () =
 		in
 		Array.fold_left (fun l fname -> get_pack_details fname @ l) [] packs
 	with _ -> []
-  
+
+let run_command cmd =
+  let ic = Unix.open_process_in cmd in
+  let result : (string list) ref = ref [] in
+  let read_str () =
+    try
+      while true do
+      	let line = input_line ic in
+      	if String.length line > 0 then result := line::(!result);
+      done
+    with _ -> () in
+  let _ = read_str() in
+  result := List.rev !result;
+  let rc = Unix.close_process_in ic in
+    (!result,rc)
+
+let get_xen_livepatch_list () = 
+	let (lines,_) = run_command ("xen-livepatch list |awk -F'|' '{gsub(/ /, \"\", $2); if ($2 == \"APPLIED\") {gsub(/[ \t]+$/, \"\", $1); print $1}}'") in
+	String.concat ", " lines
+
+let get_kpatch_list () = 
+	let (lines,_) = run_command ("kpatch list |awk 'NR>1;/^$/{exit;}'") in
```

Comment:
I think it would be better if XAPI captured the output of "kpatch list" and then processed that text rather than doing the processing by shelling out to awk.